### PR TITLE
don't throw exception in destructor in MSVC 2015

### PR DESCRIPTION
**Pull request by badbadc0ffee**
fix for issue #987

### CODE REVIEW

**Comment by KindDragon** [comment]
Related to PR #811

**Comment by badbadc0ffee** [comment]
Just realized that fixes for these issues have already been suggested back in mid-2016 e.g. in #811 - sorry for the duplicate.
Anyway: I agree that a cleaner solution would be a correct CMake configuration that correctly detects the compiler capabilities. Anyway: these minimal fixes would lead to a clean build immediately and I would suggest to file another issue for a cleanup but still accept one of the existing fixes as soon as possible...

**Comment by KindDragon** [comment]
Can be closed. PR #1218 fixed this

File: googletest/include/gtest/internal/gtest-port.h [hunk 1]
```diff
@@ -625,6 +625,7 @@ typedef struct _RTL_CRITICAL_SECTION GTEST_CRITICAL_SECTION;
 # if _MSC_VER
 #  define GTEST_HAS_HASH_MAP_ 1  // Indicates that hash_map is available.
 #  define GTEST_HAS_HASH_SET_ 1  // Indicates that hash_set is available.
+#  define _SILENCE_STDEXT_HASH_DEPRECATION_WARNINGS // Note that hash_map and hash_map are deprecated and will be removed!
```
Comment (gennadiycivil) [inline googletest/include/gtest/internal/gtest-port.h]: Shouldnt this be also checking for which version of MSVC?

**Comment by gennadiycivil** [comment]
Aplogies for not getting to this sooner and thank you for your contribution. As part of release 1.8.x preparation I am closing all older PRs that have conflicts and/or failed CI. Please resolve conflicts.

**Comment by gennadiycivil** [comment]
Aplogies for not getting to this sooner and thank you for your contribution. As part of release 1.8.x preparation I am closing all older PRs that have conflicts and/or failed CI. Please resolve conflicts.

**Pull request closed without merging**